active-rfcs/0000-vtu-deprecations.md

```diff
@@ -0,0 +1,188 @@
+# API Deprecation [RFC]
+
+Created: Mar 28, 2020 7:39 PM
+
+- Start Date: 2020.03.28
+- Target Major Version: 1.x
+- Reference Issues:
+- Implementation PR:
+
+# Summary
+
+Deprecate lesser used or bad practice methods and properties from the upcoming VueTestUtils 1.x API, aimed at Vue 3. 
+
+## Motivation
+
+With the current version of Vue Test Utils-beta.x, we notice the API is large and somewhat confusing to most  users, which are not well acquainted with testing Vue apps, leading to bad testing practices. 
+
+This PR aims to trim down the VTU API, replacing listed methods with improved docs, more examples for all the related cases or improved testing methods.
+
+## Detailed design
+
+### Methods
+
+**Wrapper.emittedByOrder -** [Link](https://vue-test-utils.vuejs.org/api/wrapper/#emittedbyorder)
+
+Rarely used, use `emitted` instead.
+
+```js
+expect(wrapper.emitted('change')[0]).toEqual(['param1', 'param2'])
+```
+
+**get -** [Link](https://vue-test-utils.vuejs.org/api/wrapper/#get) 
+
+Recently added, throws error if nothing is matched. Will be merged into `find`.
+
+```js
+expect(wrapper.find('.notExisting')).toThrow()
+
+const element = wrapper.find('.notExisting') // will throw error
+```
+
+**is -** [Link](https://vue-test-utils.vuejs.org/api/wrapper/#is)
+
+Not that much useful. Use `element.tagName` instead. 
+
+```js
+expect(wrapper.element.tagName).toEqual('div')
+```
+
+**isEmpty -** [Link](https://vue-test-utils.vuejs.org/api/wrapper/#isempty)
+
+Use custom matcher like [jest-dom](https://github.com/testing-library/jest-dom#tobeempty) on the element
+
+```js
+expect(wrapper.element).toBeEmpty()
```

Comment:
is there a way we could write `expect(wrapper).toBeEmpty()`?

Reply:
It would have to return the DOM node, instead of the VTU wrapper. So from what I saw, no.

Reply:
But couldn't we make `toBeEmpty()` read the `element` property if what's passed is a wrapper?

Reply:
If we offer people to use `jest-dom`, I think no. As that library has no connection to Vue and the VTU at all.

Reply:
Is it because `jest-dom` must be added by the user? I thought that if it's added by VTU, we could override the custom matcher that is added by them

Reply:
Is `isEmpty`  just basically sugar for `expect(wrapper.find('.foo').text())).toBe(null)`?

Reply:
it's `innerHTML === ''` so I find it quite accurate but I personally don't use it often

Reply:
I see. Let's see if any others find this method useful

Reply:
> is there a way we could write `expect(wrapper).toBeEmpty()`? (same for `isVisible`)

we'd need to extend `expect` to do so. This is the reason `jest-dom` exists as a standalone package, because it's not related to Vue Test Utils or other testing libraries, but to the test runner (in this case, Jest).

Reply:
Right, if we want anything jest specific, it may be better on a different package